File: original/DualFriendship_Easy/verifier.cpp
/*
 * @uni_kakurenbo
 * https://github.com/uni-kakurenbo/competitive-programming-workspace
 *
 * CC0 1.0  http://creativecommons.org/publicdomain/zero/1.0/deed.ja
 */
/* #language C++ GCC */
/* #region tmplate */
#include <bits/stdc++.h>
using namespace std;

#ifdef LOCAL_JUDGE
    #include<debug>
    #define debug(...) Debug::debug(Debug::split(#__VA_ARGS__), 0, __LINE__, __VA_ARGS__)
    #define DEBUG(...) do { Debug::DEBUG(nullptr, "\033[3;35m#" + to_string(__LINE__) + "\033[m  "); Debug::DEBUG(__VA_ARGS__); Debug::DEBUG(nullptr, "\033[m\n"); } while(0);
#else
    #define debug(...) { ; }
    #define DEBUG(...) { ; }
#endif

#define INF32 2147483647
#define INF64 9223372036854775807LL

#define until(...) while(!(__VA_ARGS__))

#define REP(i,n) for(int i=0, i##_length=int(n); i<i##_length; ++i)
#define REPD(i,n) for(int i=(n)-1; i>=0; --i)
#define FOR(i,a,b) for(ll i=(a), i##_last=ll(b); i<=i##_last; ++i)
#define FORD(i,a,b) for(ll i=(a), i##_last=ll(b); i>=i##_last; --i)

#define ALL(x) begin((x)),end((x))
#define RALL(x) rbegin((x)),rend((x))

#define F$ first
#define S$ second

using ll = long long;
using ull = unsigned long long;
using ld = long double;

template<class T> inline T mod(T x, T r) { return (x%r+r)%r; }

template<class T> inline bool chmax(T &a, T b) { return (a<b ? a=b, true : false); }
template<class T> inline bool chmin(T &a, T b) { return (a>b ? a=b, true : false); }
/* #endregion */

template <class T> using Graph = vector<set<T>>;

signed main() {
    int n, m; cin >> n >> m;

    assert(1 <= n and n <= 3000);
    assert(0 <= m and m <= min(n*(n-1)/2, 3000));

    Graph<int> G(n);

    REP(i, m) {
        int a, b; cin >> a >> b;
        assert(1 <= a and a < b and b <= n);
        --a, --b;
        assert(G[a].count(b) < 1 and G[a].count(b) < 1);
        G[a].emplace(b), G[b].emplace(a);
    }

    debug("OK");

    return 0;
}

File: original/tasks/TGC000/Espresso/verifier.py
# @uni_kakurenbo
# https://github.com/uni-kakurenbo/competitive-programming-workspace
#
# CC0 1.0  http://creativecommons.org/publicdomain/zero/1.0/deed.ja

# #language PyPy3 #

from sys import setrecursionlimit, stderr, argv;
# setrecursionlimit(10**5)
def debug(*args, **opts):
    if argv[-1] == "LOCAL_JUDGE": print(*args, **opts, file=stderr);

N, X, Y, K = map(int, input().split())
A = [*map(int, input().split())]

debug(N, X, Y, K)
debug(len(A))
debug(A)

if len(A) != N: print("NG: A size != N"); exit(1)
if not 1 <= N <= 4*10**6: print("NG: N"); exit(1)
if not 0 <= X <= Y <= 1000: print("NG: X, Y"); exit(1)
if Y == 0: print("NG: Y == 0"); exit(1)
if not 0 <= K <= 10**15: print("NG: K"); exit(1)
for a in A:
    if not 1 <= a <= 10**9: print("NG: A_i"); exit(1)
print("OK");
